docs/index.rst

```diff
@@ -106,23 +106,79 @@ You can use pip and make to quickly set up a development environment::
 
 Please adhere to the following guidelines:
 
-* Lint your code with flake8 and pylint. This can be done easily by executing
-  ``make lint``. The former linter must pass with zero warnings, and warnings
-  from the latter linter should be minimized.
-* Check for errors by running the unit test suite. This can be done easily by
-  executing ``make test``. It must pass with zero errors.
-* Document your code. Documentation can be easily generated by executing ``make
-  docs-html``.
-* Unit tests are *very* highly recommended.
+* All PR’s should follow the predetermined pull request template, explaining the issue, same follows for the Issue if one is to be filed.
+* Maintain Coding Standards
+    * Keep pep8 rules
+    * Follow the same stylistic and logical patterns used in the code
+        * All entity class names and class attributes have to be in the singular format
+        * All required entity attributes have to have `required=True` parameter
+        * It is preferable to use `alpha` data type for default string values for easier debug procedure
+        * In case any workaround is introduced, it is necessary to provide corresponding BZ ID directly into the code docstring
+        * All linting and formatting/style checks would be enforced by Travis-CI and PR would be considered broken until checks are passed successfully.
 * Adhere to typical commit guidelines:
-
-    * Commits should not cause NailGun to break.
+    * Commits should not cause NailGun’s unit test to fail. If it does, it will the responsibility of contributor to review those failures and fix them in the same PR's or raise another. The tracking of failures would be responsibility of contributor.
     * Commits should be small and coherent. One commit should address one issue.
     * Commits should have `good commit messages`_.
     * `Rebasing`_ is encouraged. Rebasing produces a much nicer commit history
       than merging.
+* To make the review process easy for all reviewers and anyone else interested in the new functionality please provide some output making use of your changes, this is to be added as a docstring as a example usage for the code added,  this could be in the form of a python interactive shell, test results from Robottelo or Foreman Ansible modules or soemthing else! However you intend to make use of your change will be just fine!
+* When raising a PR against any other branch which is also applicable for '6.X' or foreman branch. Please specify that information in the PR message
+* Unit tests
+    * Unit tests are compulsory
+    * Unit tests should cover all available actions, for the entity. For eg: Repository Sets, have enable, disable, list_available there should be unit tests exercising these actions.
+* When in doubt, ask on IRC. Join #robottelo on `freenode`_
+
+**Important to Note :**
+
+* Define Foreman Version labels in Nailgun
+    The moment PR would be raised, contributor needs to set the right version.
```

Comment:
Only people with write access can set labels, so not all contributors will be able to do this.

Reply:
Can we use some bot to extract the version information from PR template and set right label ?

Reply:
Yes. That's what @ansible does for example. I think we can easily extend @foremanbot to do this for us and enable it here.

Reply:
Shall we phrase this: "if possible, the contributor should set the right version" and replace it with the bot as soon as the bot works?

Reply:
ACK